### PR TITLE
Replace `github-linguist` gem with Node port `Nixinova/Linguist`

### LINKED ISSUE #424: Languages plugin no longer works
Today's metrics run deleted all languages data and doesn't add it back even in a following run. No errors are logged in the Action run.

https://github.com/Nixinova/Nixinova/commit/6339db01c52b5850557f907bf0663689b3c95812

The only difference between today and yesterday is [Linguist v7.16](https://github.com/github/linguist/releases/tag/v7.16.0) came out a few hours ago; maybe this had something to do with it?

### PR DESCRIPTION
**Pull request by Nixinova**
### Description

Using the `github-linguist` gem directly is a bit of a hack, and any changes to console.log output breaks Metrics, like what happened with the release of v7.16.

[Nixinova/Linguist](https://github.com/Nixinova/Linguist) is a Node port of `github-linguist` which allows for proper JavaScript language data generation.

This change also allows for `prose` and `data` languages to be included in any generated metrics as a new option if so desired.

### Tests
- [Generated with metrics@master](https://github.com/Nixinova/Metrics-Testing-August-1/blob/44dc92e2ebe0f88531fb9ee988da3d9cdbca4834/github-metrics.svg)
- [Generated with this change](https://github.com/Nixinova/Metrics-Testing-August-1/blob/344414271c8b8b1ce16975c0d10c2c0c1722324e/github-metrics.svg)

### Refs
- Fully resolve #424
- See also: #341

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -19,7 +19,6 @@ RUN chmod +x /metrics/source/app/action/index.mjs \
   # Based on https://github.com/github/linguist and https://github.com/github/licensed
```
Comment (Nixinova) [inline Dockerfile:19]: Whichever the Linguist imports are can be removed. Line 21?

**Review by lowlighter — APPROVED** [review]
Thanks for your contribution! 
It'll definitely improve stability of languages plugins 👍

File: Dockerfile (hunk 1 above)
Comment (lowlighter) [reply Dockerfile:19]: I don't remember which imports is for which gem unfortunately, but most of these dependencies are probably already on the base image or used somewhere else so I don't think it's too problematic to let them

Comment (Nixinova) [reply Dockerfile:19]: Linguist wants `sudo apt-get install cmake pkg-config libicu-dev zlib1g-dev libcurl4-openssl-dev libssl-dev ruby-dev` while Licensed just wants `sudo apt-get install cmake pkg-config`. I think everything from libicu-dev onwards can be removed.

**Comment by Nixinova** [comment]
Since the output of Linguist is this &darr;, is there anywhere in the languages plugin where language processing can now be simplified? (I could also add a `color` key to `languages.all[lang]` if it would be useful.)

<details><summary>Linguist output schema</summary>

```json
{
	"count": 3,
	"results": {
		"src/index.ts": "TypeScript",
		"src/cli.js": "JavaScript",
		"readme.md": "Markdown"
	},
	"languages": {
		"all": {
			"JavaScript": { "type": "programming", "bytes": 1000 },
			"TypeScript": { "type": "programming", "bytes": 2000 },
			"Markdown": { "type": "prose", "bytes": 3000 },
		},
		"programming": { "JavaScript": 1000, "TypeScript": 2000 },
		"markup": {},
		"data": {},
		"prose": { "Markdown": 3000 },
		"unknown": {},
		"total": { "unique": 3, "bytes": 6000, "unknownBytes": 0 }
	}
}
```
</details>

**Comment by lowlighter** [comment]
This loop is for GitHub api data, not for linguist data. It's the non-`indepth` mode so it cannot be removed. 

Linguist is only called in `async function analyze()` from `analyzers.mjs`, and I don't think we can simplify it further since once linguist data is gathered, that's pretty much it:
https://github.com/lowlighter/metrics/blob/63675f142161882a4dd1889fba5c875ef9979e92/source/plugins/languages/analyzers.mjs#L140-L142

Adding the `color` key would help to close #339 though, since it'll help covering languages that are not loaded from GitHub api and not user-defined

File: Dockerfile (hunk 1 above)
Comment (lowlighter) [reply Dockerfile:19]: Yes we should remove it so 👍

**Comment by lowlighter** [comment]
I'll make a release soon even if there are still improvements that can be made, since #424 still affects `@latest` release

**Comment by Nixinova** [comment]
I also think the problem in #356 may be covered by my linguist.

**Comment by Nixinova** [comment]
> Adding the `color` key would help to close #339 though, since it'll help covering languages that are not loaded from GitHub api and not user-defined

Added in 1.4.4

**Comment by lowlighter** [comment]
> I also think the problem in #356 may be covered by my linguist.

This PR was for gathering patches from GitHub events along with their associated repository `.gitattributes`. Since it's mostly for code collection I think it's unrelated though I may be wrong.

> Added in 1.4.4: get from .languages.all[lang].color

Nice 👍

File: Dockerfile (hunk 1 above)
Comment (Nixinova) [reply Dockerfile:19]: Opened #458 :+1:

**Pull request merged**